website/src/components/Visualisations/IITO/Util/ProcessTransactionData.ts

```diff
@@ -0,0 +1,162 @@
+import {
+    Link,
+    Node,
+    Transaction,
+    OutputJSONType,
+    TeamName,
+} from '../../../../consts/types'
+import { teamColors } from '../../utils'
+
+export const getIIGOTransactions = (data: OutputJSONType) => {
+    const acc: Transaction[] = []
+
+    // Since IIGOHistories is repeated, take the one from the LAST GameState and
+    // do Object.entries to make it iterable. List of array'ed tuples.
+    const IIGOHistory = Object.entries(
+        data.GameStates[data.GameStates.length - 1].IIGOHistory
+    )
+    // For each of these arrayed tuples, we have [turnNumber: <"pair events">[]]
+    IIGOHistory.forEach(([_, exchanges]) => {
+        if (exchanges) {
+            exchanges.forEach((teamAction) => {
+                const type = teamAction.Pairs[0].VariableName
+                let transaction: Transaction
+                // There are three types of transactions
+                // the target could be the client id depending on the type of team action
+                // else accounts for SanctionPaid and IslandTaxContribution
+                if (type === 'IslandAllocation') {
+                    transaction = {
+                        from: TeamName.CommonPool,
+                        to:
+                            TeamName[
+                                teamAction.ClientID as keyof typeof TeamName
+                            ],
+                        amount: teamAction.Pairs[0].Values[0],
+                    }
+                } else {
+                    transaction = {
+                        from:
+                            TeamName[
+                                teamAction.ClientID as keyof typeof TeamName
+                            ],
+                        to: TeamName.CommonPool,
+                        amount: teamAction.Pairs[0].Values[0],
+                    }
+                }
+                if (transaction.amount) acc.push(transaction)
+            })
+        }
+    })
+    return acc
+}
+
+// islandGifts should get a list of IITO Transactions that happened in that turn.
+// TODO: Try getting the newly written types to fit these functions
+export const getIITOTransactions = (data: OutputJSONType) => {
+    return data.GameStates.map((turnState) => {
```

Comment:
quite a long function that hides the complexity of two `reduce`s.

Reply:
A fair comment, but I didn't see too much point in splitting it since there's no need for the logic to be abstracted, and I annotated the function heavily, should be fine imo